uportal-war/src/main/java/org/jasig/portal/rest/PortletsRESTController.java

```diff
@@ -104,17 +107,60 @@ public ModelAndView getManageablePortlets(HttpServletRequest request, HttpServle
      */
     @RequestMapping(value="/portlet/{fname}.json", method = RequestMethod.GET)
     public ModelAndView getPortlet(HttpServletRequest request, HttpServletResponse response, @PathVariable String fname) throws Exception {
-      IPerson user = personManager.getPerson(request);
-      EntityIdentifier ei = user.getEntityIdentifier();
-      IAuthorizationPrincipal ap = AuthorizationService.instance().newPrincipal(ei.getKey(), ei.getType());
-      IPortletDefinition portletDef = portletDefinitionRegistry.getPortletDefinitionByFname(fname);
-      if(portletDef != null && ap.canRender(portletDef.getPortletDefinitionId().getStringId())) {
-        LayoutPortlet portlet = new LayoutPortlet(portletDef);
-        return new ModelAndView("json", "portlet", portlet);
-      } else {
-        response.setStatus(HttpServletResponse.SC_FORBIDDEN);
-        return new ModelAndView("json");
-      }
+        IAuthorizationPrincipal ap = getAuthorizationPrincipal(request);
+        IPortletDefinition portletDef = portletDefinitionRegistry.getPortletDefinitionByFname(fname);
+        if(portletDef != null && ap.canRender(portletDef.getPortletDefinitionId().getStringId())) {
+            LayoutPortlet portlet = new LayoutPortlet(portletDef);
+            return new ModelAndView("json", "portlet", portlet);
+        } else {
+            response.setStatus(HttpServletResponse.SC_FORBIDDEN);
+            return new ModelAndView("json");
+        }
+    }
+
+    /**
+     * Provides a single, fully-rendered portlet.  NOTE:  Access to this API
+     * enpoint requires only <code>IPermission.PORTAL_SUBSCRIBE</code> permission.
+     */
+    @RequestMapping(value="/v4-3/portlet/{fname}.html", method = RequestMethod.GET)
```

Comment:
@timlevett -- I think the thought is kinda cool, but I'm concerned that it would produce unexpected issues in practice.  We're versioning & rev-ing these URLs precisely because the accepted parameters are changing and the outputs are changing (albeit subtly).  I think it would be very difficult to write a client of one of these APIs that would be resilient to these kinds of changes.  I think you want to tie your client to a specific, known set of inputs & outputs.